src/Agent.Worker/TaskManager.cs

```diff
@@ -32,7 +32,7 @@ public interface ITaskManager : IAgentService
         void Extract(IExecutionContext executionContext, Pipelines.TaskStep task);
     }
 
-    public sealed class TaskManager : AgentService, ITaskManager
+    public class TaskManager : AgentService, ITaskManager
```

Comment:
Same question about need to derive from class.

Reply:
Still not sure what the question is, but we need this so that we can override the Load method without touching any of the rest of the functionality